### ISSUE TITLE
Command "invalidate all" fails after upgrading to 3.2

### ISSUE THREAD

**Issue by mjnaderi** [Suor/django-cacheops#242]
After I upgraded django-cacheops to 3.2, when I run command `python manage.py invalidate all`, it fails with this exception:

```
Traceback (most recent call last):
  File "/home/myuser/myproject/.venv/lib/python3.5/site-packages/funcy/decorators.py", line 49, in __getattr__
    res = self.__dict__[name] = arggetter(self._func)(name, self._args, self._kwargs)
  File "/home/myuser/myproject/.venv/lib/python3.5/site-packages/funcy/decorators.py", line 81, in get_arg
    return func.__defaults__[index - argcount]
TypeError: 'NoneType' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/home/myuser/myproject/.venv/lib/python3.5/site-packages/django/core/management/__init__.py", line 363, in execute_from_command_line
    utility.execute()
  File "/home/myuser/myproject/.venv/lib/python3.5/site-packages/django/core/management/__init__.py", line 355, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/myuser/myproject/.venv/lib/python3.5/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/myuser/myproject/.venv/lib/python3.5/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/home/myuser/myproject/.venv/lib/python3.5/site-packages/django/core/management/base.py", line 512, in handle
    label_output = self.handle_label(label, **options)
  File "/home/myuser/myproject/.venv/lib/python3.5/site-packages/cacheops/management/commands/invalidate.py", line 15, in handle_label
    self.handle_all()
  File "/home/myuser/myproject/.venv/lib/python3.5/site-packages/cacheops/management/commands/invalidate.py", line 29, in handle_all
    invalidate_all()
  File "/home/myuser/myproject/.venv/lib/python3.5/site-packages/funcy/decorators.py", line 26, in wrapper
    return deco(call, *dargs, **dkwargs)
  File "/home/myuser/myproject/.venv/lib/python3.5/site-packages/cacheops/transaction.py", line 66, in queue_when_in_transaction
    if transaction_states[call.using]:
  File "/home/myuser/myproject/.venv/lib/python3.5/site-packages/funcy/decorators.py", line 52, in __getattr__
    raise AttributeError(*e.args)
AttributeError: 'NoneType' object is not subscriptable
```

I have installed following versions of python and django:

```
Python==3.5.2
Django==1.11.1
django-cacheops==3.2
funcy==1.7.3
```

**Comment by mjnaderi** [comment]
The problem goes away with `django-cacheops==3.1.3`.

**Comment by Suor** [comment]
Can't reproduce this. This is also [a part of test suite](https://github.com/Suor/django-cacheops/blob/3.2/tox.ini#L35), so I can't see how this could be missed. I can only guess you have some broken installation, old pyc files or smth alike.

**Comment by jessicalins** [comment]
Hello @Suor,

I also had a similar error when upgrading to django-cacheops 3.2 and setting a value of a ManyToManyField:

```
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/fields/related_descriptors.py", line 843, in add
    self._add_items(self.source_field_name, self.target_field_name, *objs)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/fields/related_descriptors.py", line 1012, in _add_items
    model=self.model, pk_set=new_ids, using=db)
  File "/usr/local/lib/python2.7/dist-packages/django/dispatch/dispatcher.py", line 192, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/usr/local/lib/python2.7/dist-packages/cacheops/query.py", line 528, in invalidate_m2m
    model_column: pk
  File "/usr/local/lib/python2.7/dist-packages/funcy/decorators.py", line 25, in wrapper
    return deco(call, *dargs, **dkwargs)
  File "/usr/local/lib/python2.7/dist-packages/cacheops/transaction.py", line 66, in queue_when_in_transaction
    if transaction_states[call.using]:
  File "/usr/local/lib/python2.7/dist-packages/funcy/decorators.py", line 51, in __getattr__
    raise AttributeError(*e.args)
AttributeError: invalidate_dict() missing required argument: 'using'
```

Also when trying to use the invalidate all command:

```
Traceback (most recent call last):
  File "./manage.py", line 24, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 353, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 345, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 348, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 399, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 548, in handle
    label_output = self.handle_label(label, **options)
  File "/usr/local/lib/python2.7/dist-packages/cacheops/management/commands/invalidate.py", line 15, in handle_label
    self.handle_all()
  File "/usr/local/lib/python2.7/dist-packages/cacheops/management/commands/invalidate.py", line 29, in handle_all
    invalidate_all()
  File "/usr/local/lib/python2.7/dist-packages/funcy/decorators.py", line 25, in wrapper
    return deco(call, *dargs, **dkwargs)
  File "/usr/local/lib/python2.7/dist-packages/cacheops/transaction.py", line 66, in queue_when_in_transaction
    if transaction_states[call.using]:
  File "/usr/local/lib/python2.7/dist-packages/funcy/decorators.py", line 51, in __getattr__
    raise AttributeError(*e.args)
AttributeError: invalidate_all() missing required argument: 'using'
```

I reset my installation and also deleted the pyc files, but the error continues. I also tested using funcy vs. 1.7.1 to 1.7.4, but without success.

**Comment by Suor** [comment]
The problem occured in funcy after all, try with its latest master. Also, are you both using `CACHEOPS_DEGRADE_ON_FAILURE=True`?

**Comment by jessicalins** [comment]
It worked for me with the latest master. Thanks!

**Comment by mike86k** [comment]
I have the same problem as @jessicalins when I set `CACHEOPS_DEGRADE_ON_FAILURE=True` and trying to use the invalidate all command.
When I set `CACHEOPS_DEGRADE_ON_FAILURE=False` it looks like everything works fine.
Installing the last master didn't solve the problem.

I have installed:
```
Django==1.11.1
django-cacheops==3.2
funcy==1.7.3
```
Also I had updated `funcy` to 1.7.4 but it didn't help.

**Comment by Suor** [comment]
@jessicalins thanks for feedback.
@mike86k you misread it. It was funcy master.

To everyone. Fixed funcy is now released as 1.7.5. Fixed cacheops is 3.2.1, the only change is that forces funcy upgrade.

Fixed in 5f623da32c6e195fa7598b9843fe8188e13feeea, backported to 3.x in 34ca056.
